rllib/utils/space_utils.py

```diff
@@ -0,0 +1,53 @@
+from gym.spaces import Tuple, Dict
+
+
+def flatten_space(space):
+    """Flattens a gym.Space into its primitive components.
+
+    Primitive components are any non Tuple/Dict spaces.
+
+    Args:
+        space(gym.Space): The gym.Space to flatten. This may be any
+            supported type (including nested Tuples and Dicts).
+
+    Returns:
+        List[gym.Space]: The flattened list of primitive Spaces. This list
+            does not contain Tuples or Dicts anymore.
+    """
+
+    def _helper_flatten(space_, l):
+        if isinstance(space_, Tuple):
+            for s in space_:
+                _helper_flatten(s, l)
+        elif isinstance(space_, Dict):
+            for k in space_.spaces:
+                _helper_flatten(space_[k], l)
+        else:
+            l.append(space_)
+
+    ret = []
+    _helper_flatten(space, ret)
+    return ret
+
+
+def get_base_struct_from_space(space):
+    """
+    Returns a Tuple/Dict Space as native (and equally structured) tuple/dict.
+
+    Args:
+        space (gym.Space): The Space to get the python struct for.
+
+    Returns:
```

Comment:
Can you add an example in the documentation here?

Reply:
Sure ...

Reply:
done.